src/www/system_general.php

```diff
@@ -83,6 +84,19 @@
         $input_errors[] = gettext("The domain may only contain the characters a-z, 0-9, '-' and '.'.");
     }
 
+    /* Domain search list Validation 256 char length max */
+    if (!empty($pconfig['domainsearchlist'])) {
+      if (strlen($pconfig['domainsearchlist']) > 256) {
+        $input_errors[] = gettext("Custom domain search list can not exceed 256 char length");
+      } else {
```

Comment:
The total length of the _search_ directive cannot exceed 256. As the domain name of the system is also added to the search path and this setting is even on the same page (`$pconfig['domain']`) I'd recommend to take this into account.

```suggestion
      if (strlen($pconfig['domainsearchlist']) + strlen($pconfig['domain']) > 256) {
        $input_errors[] = gettext("Custom domain search list including the domain name must not exceed 256 characters.");
      } else {
```

Reply:
This is tricky overall. We should really have a function in system.inc that validates this so the same check/approach is used in both places (or a way to compile the final list there to check for length here).